src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/EcmaMetadata_1.cs

```diff
@@ -16,11 +17,26 @@ internal sealed class EcmaMetadata_1(Target target) : IEcmaMetadata
 
     public TargetSpan GetReadOnlyMetadataAddress(ModuleHandle handle)
     {
+        ILoader loader = target.Contracts.Loader;
         Data.Module module = target.ProcessedData.GetOrAdd<Data.Module>(handle.Address);
 
-        TargetPointer baseAddress = module.GetLoadedMetadata(out ulong size);
+        if (!loader.TryGetLoadedImageContents(handle, out _, out _, out uint imageFlags))
```

Comment:
I think I'm missing something - won't this return false and then throw for flat image layouts?

Reply:
Oh, right, nevermind - we stick the flat one in the loaded image layout slot.